signer/src/keys.rs

```diff
@@ -303,7 +319,11 @@ impl PrivateKey {
     }
     /// Creates a private key directly from a slice.
     pub fn from_slice(data: &[u8]) -> Result<Self, Error> {
-        secp256k1::SecretKey::from_slice(data)
+        if data.len() != 32 {
+            return Err(Error::InvalidPrivateKeyLength(data.len()));
+        }
+
+        secp256k1::SecretKey::from_slice(&data[0..32])
```

Comment:
I suppose this doesn't hurt, but you could just do `secp256k1::SecretKey::from_slice(data)` now.